### PR TITLE
10491: sync schemas workflow

### PR DESCRIPTION
**Pull request by mkalish**
This PR adds the workflow for syncing translation schemas between different azure steps.

Test Steps:
There is pretty comprehensive testing for this, so the easiest way to test is to uncomment `SyncTranslationSchemasTest` the code that has it run only during CI and run the test

Manual steps
1. Upload a valid and previous valid blob into the azurite-stage metadata container
2. Upload sample schemas to the azurite-stage container
3. Upload a valid and previous valid blob into the azurite metadata container
4. Run the new sync schemas command with `metadata` as the container, azurite-stage as the source and azurite as the destination
5. Wait for the validation process to complete
6. Verify that azurite contains the schema

## Changes
- A command that syncs schemas between environments
- Azure functions triggered by a validating.txt file being present that validates the schemas and rollback if an error is encountered

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #10491 
- #10494

### CODE REVIEW

File: prime-router/src/main/kotlin/azure/BlobAccess.kt [hunk 1]
```diff
@@ -356,6 +360,26 @@ class BlobAccess() : Logging {
...
+            val currentBlobClient = container.getBlobClient(blobItemAndPreviousVersions.currentBlobItem.name)
+            val previousVersionId = blobItemAndPreviousVersions.previousBlobItemVersions?.firstOrNull()?.versionId
+            if (previousVersionId != null) {
+                val previousVersionBlobClient = container.getBlobVersionClient(
+                    blobItemAndPreviousVersions.currentBlobItem.name,
+                    previousVersionId
+                )
+                currentBlobClient.copyFromUrl(previousVersionBlobClient.blobUrl)
+            } else {
+                logger.error(
```
Comment (mkalish) [inline prime-router/src/main/kotlin/azure/BlobAccess.kt:377]: I toyed with throwing and exception here but ultimately decided it was not necessary and there wasn't anything to recover.

File: prime-router/src/main/kotlin/azure/BlobAccess.kt [hunk 2]
```diff
@@ -397,7 +427,7 @@ class BlobAccess() : Logging {
          * If one exists for the container name and connection string, the existing one will be reused.
          * @return the blob container client
          */
-        private fun getBlobContainer(blobConnInfo: BlobContainerMetadata): BlobContainerClient {
+        internal fun getBlobContainer(blobConnInfo: BlobContainerMetadata): BlobContainerClient {
```
Comment (mkalish) [inline prime-router/src/main/kotlin/azure/BlobAccess.kt:430]: Exposed to it can get mocked in tests

**Review by arnejduranovic — COMMENTED** [review]
Looks great! Waiting for final version to approve

File: prime-router/build.gradle.kts [hunk 3]
```diff
@@ -442,6 +442,13 @@ tasks.register("testS2S") {
     finalizedBy("primeCLI")
 }
 
+tasks.register("testSyncS") {
```
Comment (mkalish) [inline prime-router/build.gradle.kts:445]: Temporary for easier testing, I'll remove before merging.

File: .github/actions/build-backend/action.yml [hunk 4]
```diff
@@ -94,7 +93,7 @@ runs:
       if: inputs.run-integration-tests == 'true'
       working-directory: prime-router
       id: spin_containers
-      run: docker-compose -f docker-compose.yml up -d prime_dev sftp azurite
+      run: docker-compose -f docker-compose.yml up -d prime_dev sftp azurite azurite-stage
```
Comment (mkalish) [inline .github/actions/build-backend/action.yml:96]: This is the only real change, everything else is formatting.

**Review by snesm — APPROVED** [review]

**Review by arnejduranovic — APPROVED** [review]
Changes and tests look really good. Verified the new functional tests run successfully in CI.

**Pull request merged**